Update HTTPS configuration

Angristan opened the pull request:
We should disable DH and use AES CBC/GCM + ECDH ciphers only. It's fast and secure, and widely compatible (cf https://tls.imirhil.fr/suite).
Also, HSTS should not be enabled by default, I think HSTS should be activated manually while knowing what are its consequences.

SamSaffron commented:
I strongly disagree with dropping HSTS by default, @mpalmer's call about changing anything with our cipher suite.

mpalmer commented:
Yeah, HSTS is where it's at.  Removing that significantly reduces security, and I'm not in favour of it.

As far as changing the cipher suites goes, what are the *actual* benefits of this change?  Is there one or more supported user agents which, with the current cipher suite set, select a sub-optimal cipher suite?  Also, I'll need to see what an ssllabs scan looks like for both the current and new configurations to decide whether whatever changes is acceptable.  If you want this change to happen before I get around to spinning up a scratch machine to point ssllabs at, you'll probably want to do those tests and paste the results here.

Angristan commented:
I'm okay with HSTS, I changed my mind.

I changed the cipher suite to show which ciphers are used.

With the new one we drop DH, 3DES, Camellia and ECDSA certificates because Let's Encrypt gives RSA certificates. We only use ECDH + RSA + AES 128/256 GCM/CBC + SHA/SHA256/SHA384.

3DES is weak, so we have to drop it. DH is ok, ECDH is faster and stronger, and does not require to generate a key. We use a secp384r1 ECDH key instead. There is no point to support Camellia because it does not add any compatibility.

Current configuration : 
![screenshot_09-02-2017_b hwdf36cj](https://cloud.githubusercontent.com/assets/11699655/22787247/f6133b9e-eedb-11e6-8648-864c379bc3f7.png)

![screenshot_09-02-2017_b rchv_dus](https://cloud.githubusercontent.com/assets/11699655/22787263/fce03170-eedb-11e6-9188-fc236252f504.png)

New configuration : 
![screenshot_09-02-2017_b idnuuj7o](https://cloud.githubusercontent.com/assets/11699655/22787290/0f11d830-eedc-11e6-958b-8e61b55317ef.png)

![screenshot_09-02-2017_b m_otd-bv](https://cloud.githubusercontent.com/assets/11699655/22789611/c36bd77a-eee3-11e6-8e1f-94d1b7360fde.png)


According to SSLLabs, it removes support for : 
- IE 6 / XP
- IE 8 / XP
- Java 6u45 
- OpenSSL 0.9.8y
- Android 2.3.7

So only very old systems.

This PR is not a major security upgrade, just a little improvement. :)

mpalmer commented:
Thanks for the testing and details about compatibility.  Merged!

xfalcox commented:
This will make installs even faster, thanks @Angristan!

robbyoconnor commented:
I was _ALMOST_ objecting to this but nevermind :) Good work!
